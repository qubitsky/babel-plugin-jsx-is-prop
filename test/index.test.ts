import { pluginTester } from "babel-plugin-tester";
import isPropPlugin from "../src/index";

pluginTester({
  plugin: isPropPlugin,
  pluginName: 'jsxIsPropTransformPlugin',
  babelOptions: {
    plugins: [["@babel/plugin-syntax-jsx"]],
  },
  snapshot: true,
  tests: [
    {
      code: "<foo is={Bar} />",
    },
    {
      code: "<foo is={Bar} hello={world} />",
    },
  ],
});
