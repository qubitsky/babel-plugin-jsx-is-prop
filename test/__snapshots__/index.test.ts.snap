// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jsxIsPropTransformPlugin 1. jsxIsPropTransformPlugin: 1. jsxIsPropTransformPlugin 1`] = `

<foo is={Bar} />

      ↓ ↓ ↓ ↓ ↓ ↓

<Bar />;

`;

exports[`jsxIsPropTransformPlugin 2. jsxIsPropTransformPlugin: 2. jsxIsPropTransformPlugin 1`] = `

<foo is={Bar} hello={world} />

      ↓ ↓ ↓ ↓ ↓ ↓

<Bar hello={world} />;

`;
